File: android/app/src/main/java/com/example/socialfeed/data/db/entity/Like.kt
package com.example.socialfeed.data.db.entity

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index

@Entity(
    tableName = "likes",
    primaryKeys = ["postId", "userId"],
    foreignKeys = [
        ForeignKey(entity = Post::class, parentColumns = ["id"], childColumns = ["postId"], onDelete = ForeignKey.CASCADE),
        ForeignKey(entity = User::class, parentColumns = ["id"], childColumns = ["userId"], onDelete = ForeignKey.CASCADE)
    ],
    indices = [Index("postId"), Index("userId")]
)
data class Like(
    val postId: String,
    val userId: String
)

File: backend-cs/Models/Like.cs
namespace SocialFeed.Api.Models;

public class Like
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public string PostId { get; set; } = string.Empty;
    public string UserId { get; set; } = string.Empty;

    public Post Post { get; set; } = null!;
    public User User { get; set; } = null!;
}
